torch/csrc/jit/serialization/pickler.cpp

```diff
@@ -801,4 +801,92 @@ bool checkHasValidSetGetState(const std::shared_ptr<c10::ClassType>& cls) {
+using BackendMetaPtr =
+    void (*)(const at::Tensor&, std::unordered_map<std::string, bool>&);
+
+// map to save function pointer for BackendMeta serialization.
+static std::unordered_map<std::string, void*> fptr_map;
+
+// Register function pointer of Tensor BackendMetadata for serialization.
+void TensorBackendMetaRegistry(bool is_set, void* fptr) {
+  if (is_set) {
+    TORCH_CHECK(
+        fptr_map.find("set") == fptr_map.end(),
+        "The function that set tensor BackendMeta for deserialization has been registered.");
+
+    fptr_map["set"] = fptr;
+  } else {
+    TORCH_CHECK(
+        fptr_map.find("get") == fptr_map.end(),
+        "The function that get tensor BackendMeta for serialization has been registered.");
+
+    fptr_map["get"] = fptr;
+  }
+}
+
+// Return a map of Tensor Metadata which including BackendMetaData for
+// serialization. For now, it only takes care of `conj` and `neg` bit.
+std::unordered_map<std::string, bool> getTensorMetadata(const at::Tensor& t) {
+  // We don't support serializing `ZeroTensor` as it is not public
+  // facing yet.
+  TORCH_CHECK(
+      !t._is_zerotensor(),
+      "ZeroTensor is not serializable,",
+      " please file an issue if required.");
+  std::unordered_map<std::string, bool> metadata{};
+
+  // Only add meta-data if the value is not default.
+  if (t.is_conj()) {
+    metadata["conj"] = true;
+  }
+  if (t.is_neg()) {
+    metadata["neg"] = true;
+  }
+  // Only add BackendMetaData for custom backend if the function pointer is
+  // registered.
+  if (fptr_map.find("get") != fptr_map.end() &&
+      t.device().type() == c10::DeviceType::PrivateUse1) {
+    // Pass the tensor and metadata map references as parameters to the custom
+    // serialization function.
+    ((BackendMetaPtr)fptr_map["get"])(t, metadata);
+  }
+  return metadata;
+}
+
+void setTensorMetadata(
+    const at::Tensor& t,
+    std::unordered_map<std::string, bool> metadata) {
+  auto iter_end = metadata.end();
+  auto iter_temp = metadata.find("conj");
+  if (iter_temp != iter_end) {
+    t._set_conj(true);
+    metadata.erase(iter_temp);
+  }
+  iter_temp = metadata.find("neg");
+  if (iter_temp != iter_end) {
+    t._set_neg(true);
+    metadata.erase(iter_temp);
+  }
+  // Only set BackendMetaData for custom backend if the function pointer is
+  // registered.
+  if (fptr_map.find("set") != fptr_map.end() &&
+      t.device().type() == c10::DeviceType::PrivateUse1) {
+    // Pass the tensor and metadata map references as parameters to the custom
+    // deserialization function.
+    ((BackendMetaPtr)fptr_map["set"])(t, metadata);
+  }
+}
+
+void setTensorMetadata(
+    const at::Tensor& t,
+    c10::Dict<c10::IValue, c10::IValue> metadata_idict) {
+  std::unordered_map<std::string, bool> metadata;
+  for (auto& pair : metadata_idict) {
+    auto key = *pair.key().toString();
+    metadata[key] = pair.value().toBool();
+  }
+  setTensorMetadata(t, std::move(metadata));
```

Comment:
no change?

Reply:
This function has overloading, I made a change in the function that will eventually be called
`void setTensorMetadata(const at::Tensor& t, std::unordered_map<std::string, bool> metadata) `， at line 857